tests/static-analysis/assert-ipv4.php

```diff
@@ -0,0 +1,17 @@
+<?php
+
+namespace Webmozart\Assert\Tests\StaticAnalysis;
+
+use Webmozart\Assert\Assert;
+
+/**
+ * @psalm-param non-empty-string $value
```

Comment:
```suggestion
 * @psalm-pure
 * @psalm-param non-empty-string $value
```

Reply:
Sadly not: the various `filter_var` usages are not pure when combined with `mixed` arguments, as they can operate on objects implementing `__toString`